Sources/MongoSwift/MongoClient.swift

```diff
@@ -150,6 +150,7 @@ public class MongoClient {
         if options?.eventMonitoring == true { self.initializeMonitoring() }
 
         guard mongoc_client_set_error_api(self._client, MONGOC_ERROR_API_VERSION_2) else {
+            self.close()
```

Comment:
Is this done in other cases, such as failure to initialize monitoring?

Was this added due to code analysis, or would we actually expect `mongoc_client_set_error_api()` to fail in a real use case?

Reply:
As the driver is now, this call should never fail. If we change our threading/client pooling model, it will (you need to invoke this on the pool rather than a client derived from the pool). I added the check so that if we did start using client pools, tests would break unless this call was updated.